src/sage/categories/examples/semirings.py

```diff
@@ -0,0 +1,264 @@
+            return False
+        return self._n == other._n
+
+    def __ne__(self, other):
+        """
+        Test for non-equality.
+
+        TESTS::
+
+            sage: from sage.categories.examples.semirings import TernaryLogic
+            sage: S = TernaryLogic()
+            sage: S(1) != S(2)
+            True
+        """
+        return not (self == other)
+
+
+class TernaryLogic(UniqueRepresentation, Parent):
+    r"""
+    An example of a semiring.
+
+    This class illustrates a minimal implementation of a semiring.
+
+    EXAMPLES::
+
+        sage: S = Semirings().example(); S
+        An example of a semiring: the ternary-logic semiring
+
+    This is the semiring that contains 3 objects::
+
+        sage: S.some_elements()
+        [0, 1, many]
+
+    The product rule is as expected::
+
+        sage: S(1) * S(1)
+        1
+        sage: S(1) + S(1)
+        many
+
+    TESTS::
+
+        sage: TestSuite(S).run(verbose=True)
+        running ._test_additive_associativity() . . . pass
+        running ._test_an_element() . . . pass
+        running ._test_associativity() . . . pass
+        running ._test_cardinality() . . . pass
+        running ._test_category() . . . pass
+        running ._test_construction() . . . pass
+        running ._test_distributivity() . . . pass
+        running ._test_elements() . . .
+          Running the test suite of self.an_element()
+          running ._test_category() . . . pass
+          running ._test_eq() . . . pass
+          running ._test_new() . . . pass
+          running ._test_nonzero_equal() . . . pass
+          running ._test_not_implemented_methods() . . . pass
+          running ._test_pickling() . . . pass
+          pass
+        running ._test_elements_eq_reflexive() . . . pass
+        running ._test_elements_eq_symmetric() . . . pass
+        running ._test_elements_eq_transitive() . . . pass
+        running ._test_elements_neq() . . . pass
+        running ._test_eq() . . . pass
+        running ._test_new() . . . pass
+        running ._test_not_implemented_methods() . . . pass
+        running ._test_one() . . . pass
+        running ._test_pickling() . . . pass
+        running ._test_prod() . . . pass
+        running ._test_some_elements() . . . pass
+        running ._test_zero() . . . pass
+    """
+    def __init__(self):
+        r"""
+        The ternary-logic semiring.
+
+        EXAMPLES::
+
+            sage: S = Semirings().example(); S
+            An example of a semiring: the ternary-logic semiring
+        """
+        Parent.__init__(self, category=Semirings())
+
+    def _repr_(self):
+        r"""
+        Return the string representation.
+
+        EXAMPLES::
+
+            sage: Semirings().example()._repr_()
+            'An example of a semiring: the ternary-logic semiring'
+        """
+        return "An example of a semiring: the ternary-logic semiring"
+
+    def __contains__(self, n) -> bool:
```

Comment:
I am not sure this is necessary from the default one in `Parent`. Can you check?